opm/autodiff/BlackoilModelEbos.hpp

```diff
@@ -221,24 +206,29 @@ namespace Opm {
                                            ReservoirState& reservoir_state,
                                            WellState& well_state)
         {
-            const double dt = timer.currentStepLength();
             if (iteration == 0) {
                 // For each iteration we store in a vector the norms of the residual of
                 // the mass balance for each active phase, the well flux and the well equations.
                 residual_norms_history_.clear();
                 current_relaxation_ = 1.0;
-                dx_old_ = V::Zero(sizeNonLinear());
+                dx_old_ = 0.0;
             }
             IterationReport iter_report = assemble(timer, iteration, reservoir_state, well_state);
-            residual_norms_history_.push_back(computeResidualNorms());
-            const bool converged = getConvergence(timer, iteration);
-            const bool must_solve = (iteration < nonlinear_solver.minIter()) || (!converged);
+            std::vector<double> residual_norms;
+            const bool converged = getConvergence(timer, iteration,residual_norms);
+            residual_norms_history_.push_back(residual_norms);
+            const bool must_solve = true;
+            Dune::InverseOperatorResult result;
             if (must_solve) {
```

Comment:
Ops. The current version solves the linear system even if the solution is converged. I will push a fix.